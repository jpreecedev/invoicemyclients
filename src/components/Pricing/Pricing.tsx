import React from 'react'
import { Link } from 'react-router-dom'

import { ROUTES } from '../../routes'

const Pricing = () => {
  return (
    <section className="py-8 mt-10 mb-10">
      <div className="container">
        <div className="row justify-content-center">
          <div className="col-12 col-md-5">
            <div className="card shadow-lg mb-6 mb-md-0">
              <div className="card-body">
                <div className="text-center mb-3">
                  <span className="badge badge-pill badge-primary-soft">
                    <span className="h6 text-uppercase">Free</span>
                  </span>
                </div>

                <div className="d-flex justify-content-center">
                  <span className="price display-2 mb-0">$0.00</span>
                </div>

                <p className="text-center text-muted mb-5">Free Forever</p>

                <div className="d-flex">
                  <div className="badge badge-rounded-circle badge-success-soft mt-1 mr-4">
                    <i className="fe fe-check"></i>
                  </div>

                  <p>Every feature, free for life.</p>
                </div>
                <div className="d-flex">
                  <div className="badge badge-rounded-circle badge-success-soft mt-1 mr-4">
                    <i className="fe fe-check"></i>
                  </div>

                  <p className="mb-5">Never pay a penny.</p>
                </div>

                <Link to={ROUTES.signup} className="btn btn-block btn-primary">
                  Sign Up Now <i className="fe fe-arrow-right ml-3"></i>
                </Link>
              </div>
            </div>
          </div>
          <div className="col-12 col-md-5">
            <div className="card shadow-lg mb-md-0">
              <div className="card-body">
                <div className="text-center mb-3">
                  <span className="badge badge-pill badge-primary-soft">
                    <span className="h6 text-uppercase">Pay As You Get Paid</span>
                  </span>
                </div>

                <div className="d-flex justify-content-center">
                  <span className="price display-2 mb-0" data-annual="49" data-monthly="69">
                    From 2.4%
                  </span>
                </div>

                <p className="text-center text-muted mb-5">Pay when your client pays you</p>

                <div className="d-flex">
                  <div className="badge badge-rounded-circle badge-success-soft mt-1 mr-4">
                    <i className="fe fe-check"></i>
                  </div>

                  <p>2.4% for European Payments</p>
                </div>
                <div className="d-flex">
                  <div className="badge badge-rounded-circle badge-success-soft mt-1 mr-4">
                    <i className="fe fe-check"></i>
                  </div>

                  <p className="mb-5">3.9% for International Payments</p>
                </div>

                <Link to={ROUTES.signup} className="btn btn-block btn-primary-soft">
                  Sign Up Now <i className="fe fe-arrow-right ml-3"></i>
                </Link>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  )
}

export { Pricing }
