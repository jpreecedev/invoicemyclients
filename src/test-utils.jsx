import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import { render } from '@testing-library/react'
import { createMemoryHistory } from 'history'

function renderWithRouter(
  ui,
  { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
  const Wrapper = ({ children }) => <Router history={history}>{children}</Router>
  return {
    ...render(ui, { wrapper: Wrapper }),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history
  }
}

export { renderWithRouter }
