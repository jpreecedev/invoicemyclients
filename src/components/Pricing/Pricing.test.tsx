import React from 'react'
import { renderWithRouter } from '../../test-utils'

import { Pricing } from './Pricing'

describe('<Pricing /> tests', () => {
  it('should match snapshot', () => {
    expect(renderWithRouter(<Pricing />).container).toMatchSnapshot()
  })
})
