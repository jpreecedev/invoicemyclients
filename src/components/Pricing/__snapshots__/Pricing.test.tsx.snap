// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pricing /> tests should match snapshot 1`] = `
<div>
  <section
    class="py-8 mt-10 mb-10"
  >
    <div
      class="container"
    >
      <div
        class="row justify-content-center"
      >
        <div
          class="col-12 col-md-5"
        >
          <div
            class="card shadow-lg mb-6 mb-md-0"
          >
            <div
              class="card-body"
            >
              <div
                class="text-center mb-3"
              >
                <span
                  class="badge badge-pill badge-primary-soft"
                >
                  <span
                    class="h6 text-uppercase"
                  >
                    Free
                  </span>
                </span>
              </div>
              <div
                class="d-flex justify-content-center"
              >
                <span
                  class="price display-2 mb-0"
                >
                  $0.00
                </span>
              </div>
              <p
                class="text-center text-muted mb-5"
              >
                Free Forever
              </p>
              <div
                class="d-flex"
              >
                <div
                  class="badge badge-rounded-circle badge-success-soft mt-1 mr-4"
                >
                  <i
                    class="fe fe-check"
                  />
                </div>
                <p>
                  Every feature, free for life.
                </p>
              </div>
              <div
                class="d-flex"
              >
                <div
                  class="badge badge-rounded-circle badge-success-soft mt-1 mr-4"
                >
                  <i
                    class="fe fe-check"
                  />
                </div>
                <p
                  class="mb-5"
                >
                  Never pay a penny.
                </p>
              </div>
              <a
                class="btn btn-block btn-primary"
                href="/sign-up"
              >
                Sign Up Now 
                <i
                  class="fe fe-arrow-right ml-3"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          class="col-12 col-md-5"
        >
          <div
            class="card shadow-lg mb-md-0"
          >
            <div
              class="card-body"
            >
              <div
                class="text-center mb-3"
              >
                <span
                  class="badge badge-pill badge-primary-soft"
                >
                  <span
                    class="h6 text-uppercase"
                  >
                    Pay As You Get Paid
                  </span>
                </span>
              </div>
              <div
                class="d-flex justify-content-center"
              >
                <span
                  class="price display-2 mb-0"
                  data-annual="49"
                  data-monthly="69"
                >
                  From 2.4%
                </span>
              </div>
              <p
                class="text-center text-muted mb-5"
              >
                Pay when your client pays you
              </p>
              <div
                class="d-flex"
              >
                <div
                  class="badge badge-rounded-circle badge-success-soft mt-1 mr-4"
                >
                  <i
                    class="fe fe-check"
                  />
                </div>
                <p>
                  2.4% for European Payments
                </p>
              </div>
              <div
                class="d-flex"
              >
                <div
                  class="badge badge-rounded-circle badge-success-soft mt-1 mr-4"
                >
                  <i
                    class="fe fe-check"
                  />
                </div>
                <p
                  class="mb-5"
                >
                  3.9% for International Payments
                </p>
              </div>
              <a
                class="btn btn-block btn-primary-soft"
                href="/sign-up"
              >
                Sign Up Now 
                <i
                  class="fe fe-arrow-right ml-3"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
